test_scripts/RC/ButtonPress/010_generic_error_if_no_response_from_hmi.lua

```diff
@@ -0,0 +1,61 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/smartdevicelink/sdl_requirements/issues/9
+-- Use case: https://github.com/smartdevicelink/sdl_requirements/blob/master/detailed_docs/button_press_emulation.md
+-- Item: Use Case 1: Main Flow
+--
+-- Requirement summary:
+-- [SDL_RC] Button press event emulation
+--
+-- Description:
+-- In case:
+-- 1) RC app sends ButtonPress request with valid parameters
+-- 2) and HMI didn't respond within default timeout
+-- SDL must:
+-- 1) Respond to App with success:false, "GENERIC_ERROR"
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local commonRC = require('test_scripts/RC/commonRC')
+local commonTestCases = require('user_modules/shared_testcases/commonTestCases')
+
+--[[ Local Variables ]]
+local modules = { "CLIMATE", "RADIO" }
+
+--[[ Local Functions ]]
+local function getDataForModule(pModuleType, self)
+  local cid = self.mobileSession1:SendRPC("ButtonPress", {
+    moduleType = pModuleType,
+    buttonName = commonRC.getButtonNameByModule(pModuleType),
+    buttonPressMode = "SHORT"
+  })
+
+  EXPECT_HMICALL("Buttons.ButtonPress", {
+    appID = self.applications["Test Application"],
+    moduleType = pModuleType,
+    buttonName = commonRC.getButtonNameByModule(pModuleType),
+    buttonPressMode = "SHORT"
+  })
+  :Do(function(_, _)
+    -- HMI does not respond
+    end)
+
+  self.mobileSession1:ExpectResponse(cid, { success = false, resultCode = "GENERIC_ERROR"})
```

Comment:
Is a timeout needed here since the default timeout is the same as SDL Core's?

Reply:
If timed wrong, the test could fail because the ATF timeout might fire before Core sends this response.

Reply:
Never mind, didn't see the `DelayedExp()` call